[BUG / TYPESCRIPT] `vue-tsc` flags auto-imports as type errors

## Description
As of PR #724 being merged to `staging` we now have type-checking occuring on our `.vue` files. There are piles of type-errors to address, but one that keeps popping up is **error TS2304: Cannot find name '[NAME]'**

Running `npx vue-tsc` at the command line and the first error you see is:

```
components/ChallengeRender.vue:8:20 - error TS2304: Cannot find name 'computed'
const experience = computed(() => {
```
The `computed()` method from the Vue Reactivity API is auto-imported by Nuxt whenever it is used in a `.vue` file. In conventional Nuxt syntax, using an `import` statement for these methods has no effect. Composables and components work similarly. However, `vue-tsc` doesn't know this and throws an error because it can't find the function definition.

Read more about Nuxt auto-imports [here](https://nuxt.com/docs/4.x/guide/concepts/auto-imports) 

## Ideal Solution
Update the `vue-tsc` configuration so that it doesn't flag Nuxt auto-imports as type errors.

This issue could theoretically be solved by adding in import statements to our `.vue` files, but this seems like a bad idea. It will bloat our code, introduce un-idiomatic syntax into our project, and all to make a mis-configured type-checker happy.

## Reproduction

Fork the `staging` branch and run `npx vue-tsc`. Note all the `error TS2304: Cannot find name ...` errors. Spin up the test server and notice that Nuxt has no problem resolving these imports.

--- a/app/components/EncounterBuilder.vue
+++ b/app/components/EncounterBuilder.vue
@@ -168,14 +168,6 @@
 </template>
 
 <script setup lang="ts">
-import { ref, computed, onMounted } from 'vue';
-import {
-  useEncounterStore,
-  type DifficultyLevel,
-} from '~/composables/useEncounter';
-import { usePartyStore } from '~/composables/useParty';
-import EncounterBuilderPartyBuilder from '~/components/EncounterBuilderPartyBuilder.vue';
-import EncounterBuilderMonsterSearch from '~/components/EncounterBuilderMonsterSearch.vue';
 import type { Monster } from '~/types/monster';
 
 // Prop included for testing purposes

--- a/app/components/EncounterBuilder.vue
+++ b/app/components/EncounterBuilder.vue
@@ -168,6 +168,7 @@
 </template>
 
 <script setup lang="ts">
+import EncounterBuilderMonsterSearch from '~/components/EncounterBuilderMonsterSearch.vue';
 import type { Monster } from '~/types/monster';
 
 // Prop included for testing purposes

--- a/app/components/EncounterBuilderPartyBuilder.vue
+++ b/app/components/EncounterBuilderPartyBuilder.vue
@@ -47,9 +47,6 @@
 </template>
 
 <script setup lang="ts">
-import { usePartyStore } from '~/composables/useParty';
-import { onMounted } from 'vue';
-
 const store = usePartyStore();
 const { partyRows, addPartyRow, removePartyRow }
   = store;

--- a/app/components/Modal.vue
+++ b/app/components/Modal.vue
@@ -86,7 +86,6 @@
 </template>
 
 <script setup>
-import { useSlots } from 'vue';
 import {
   Dialog,
   DialogPanel,

--- a/app/components/ModalReportIssue.vue
+++ b/app/components/ModalReportIssue.vue
@@ -17,8 +17,6 @@
 </script>
 
 <script setup>
-import { ref } from 'vue';
-
 const isOpen = ref(false);
 const formData = ref({});
 const status = ref('ready');

--- a/app/components/ResultsTableRow.vue
+++ b/app/components/ResultsTableRow.vue
@@ -51,8 +51,6 @@
 </template>
 
 <script setup>
-import SourceTag from './SourceTag.vue';
-
 defineProps({
   data: { type: Object, default: () => {} }, // Open5e data to render
   cols: { type: Array, default: () => [] }, // Arr. of table columns to render

--- a/app/composables/useFindPaginated.ts
+++ b/app/composables/useFindPaginated.ts
@@ -20,7 +20,7 @@ export const useFindPaginated = (options: {
   sortByProperty?: MaybeRef<string>;
   isSortDescending?: MaybeRef<boolean>;
   filter?: MaybeRef<Record<string, never>>;
-  params?: MaybeRef<Record<string, never>>;
+  params?: MaybeRef<Record<string, string | number | boolean>>;
 }) => {
   const {
     endpoint,

--- a/app/pages/monsters/[id].vue
+++ b/app/pages/monsters/[id].vue
@@ -339,8 +339,6 @@
 </template>
 
 <script setup lang="ts">
-import { useEncounterStore } from '~/composables/useEncounter';
-
 const route = useRoute();
 const params = {
   environments__fields: 'name',
@@ -381,7 +379,7 @@ const actions = computed(() => {
 });
 
 // Converts SNAKE_CASE to Title Case, used for action type headers
-const snakeToTitleCase = input =>
+const snakeToTitleCase = (input: string) =>
   input
     .toLowerCase()
     .split('_')

--- a/app/pages/monsters/index.vue
+++ b/app/pages/monsters/index.vue
@@ -140,10 +140,9 @@
 </template>
 
 <script setup lang="ts">
-import { h, computed } from 'vue';
+import { h } from 'vue';
 import { PlusIcon, MinusIcon } from '@heroicons/vue/24/solid';
 import type { Monster } from '~/types/monster';
-import { useEncounterStore } from '~/composables/useEncounter';
 
 // Set up filters
 const filterState = useFilterState<MonsterFilter>({

--- a/app/pages/search/index.vue
+++ b/app/pages/search/index.vue
@@ -66,8 +66,6 @@
 </template>
 
 <script setup>
-import { computed } from 'vue';
-
 const searchText = useQueryParam('text');
 const { data } = useSearch(searchText);
 const { sources } = useSourcesList();

--- a/nuxt.config.ts
+++ b/nuxt.config.ts
@@ -96,7 +96,12 @@ export default defineNuxtConfig({
   },
 
   typescript: {
-    strict: false,
+    // typeCheck: true,
+    strict: true
+  },
+  imports: {
+    autoImport: true,
+    global: true,
   },
 
   compatibilityDate: '2024-11-16',

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,31 +1,6 @@
 {
+  "extends": "./.nuxt/tsconfig.json",
   "compilerOptions": {
-    "target": "esnext",
-    "module": "esnext",
-    "moduleResolution": "node",
-    "lib": ["esnext", "dom"],
-    "jsx": "preserve",
-    "baseUrl": ".",
-    "paths": {
-      "~/*": ["./app/*"],
-      "@/*": ["./app/*"]
-    },
-    "types": ["@types/node", "@nuxt/types"],
-    "esModuleInterop": true,
-    "noEmit": true,
-    "forceConsistentCasingInFileNames": true,
-    "strict": true,
-    "skipLibCheck": true
+    "types": ["@types/node"]
   },
-  "include": [
-    "nuxt.config.ts",
-    // TODO: `vue-tsc` required to type-check .vue files (issue #714)
-    "pages/**/*.vue",
-    "layouts/default.vue",
-    "components/**/*.vue",
-    "composables/**/*.ts",
-    "plugins/**/*.ts",
-    "store/**/*.ts"
-  ],
-  "exclude": ["node_modules"]
 }

